### PR TITLE
Add GlyphsRasterizationMode and API

### PR DESCRIPTION
**Pull request by Chaoba**
This pr add GlyphsRasterizationMode for client-side text rendering.

### CODE REVIEW

File: MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/MapboxMapOptionsAttrsTest.kt [hunk 1]
```diff
@@ -75,6 +75,30 @@ class MapboxMapOptionsAttrsTest {
...
+    mockEnableLocalIdeograph(enabled = true)
+    val font = "foo"
+    mockLocalIdeographString(font)
+    every { typedArray.getInt(R.styleable.mapbox_MapView_mapbox_glyphRasterizationMode, 0) } returns 1
+    val options = MapboxMapOptions.createFromAttributes(options, context, typedArray)
+    verify(exactly = 1) {
+      options.setRasterizationMode(GlyphsRasterizationMode.IDEOGRAPHS_RASTERIZED_LOCALLY)
+    }
+  }
+
```
Comment (kiryldz) [inline MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/MapboxMapOptionsAttrsTest.kt:101]: Similar test for `IDEOGRAPHS_RASTERIZED_LOCALLY` mode?

**Review by kiryldz — APPROVED** [review]

File: MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java [hunk 2]
```diff
@@ -82,6 +82,7 @@
   private boolean localIdeographFontFamilyEnabled = true;
   private String localIdeographFontFamily;
   private String[] localIdeographFontFamilies;
+  private GlyphsRasterizationMode glyphsRasterizationMode = GlyphsRasterizationMode.NO_GLYPHS_RASTERIZED_LOCALLY;
```
Comment (tobrun) [inline MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java:85]: For hydrogen the default is generating the CJKV langauges locally

Comment (tobrun) [reply MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java:85]: concretely IDEOGRAPHS_RASTERIZED_LOCALLY

File: MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/MapboxMapOptionsAttrsTest.kt (hunk 1 above)
Comment (Chaoba) [reply MapboxGLAndroidSDK/src/test/java/com/mapbox/mapboxsdk/maps/MapboxMapOptionsAttrsTest.kt:101]: I guess is `ALL_GLYPHS_RASTERIZED_LOCALLY`, right?

**Review by tobrun — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Pull request merged**